File: app/src/main/java/eu/darken/bluemusic/settings/ui/SettingsActivity.java
package eu.darken.bluemusic.settings.ui;

import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v7.app.AppCompatActivity;

import javax.inject.Inject;

import butterknife.ButterKnife;
import eu.darken.bluemusic.R;
import eu.darken.bluemusic.settings.ui.general.SettingsFragment;
import eu.darken.mvpbakery.base.MVPBakery;
import eu.darken.mvpbakery.base.ViewModelRetainer;
import eu.darken.mvpbakery.injection.ComponentSource;
import eu.darken.mvpbakery.injection.InjectedPresenter;
import eu.darken.mvpbakery.injection.ManualInjector;
import eu.darken.mvpbakery.injection.PresenterInjectionCallback;
import eu.darken.mvpbakery.injection.fragment.HasManualFragmentInjector;


public class SettingsActivity extends AppCompatActivity implements SettingsActivityPresenter.View, HasManualFragmentInjector {

    @Inject ComponentSource<Fragment> componentSource;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        setTheme(R.style.BaseAppTheme);
        super.onCreate(savedInstanceState);
        MVPBakery.<SettingsActivityPresenter.View, SettingsActivityPresenter>builder()
                .presenterFactory(new InjectedPresenter<>(this))
                .presenterRetainer(new ViewModelRetainer<>(this))
                .addPresenterCallback(new PresenterInjectionCallback<>(this))
                .attach(this);

        setContentView(R.layout.activity_layout_settings);
        ButterKnife.bind(this);
    }

    @Override
    public void showSettings() {
        Fragment introFragment = getSupportFragmentManager().findFragmentById(R.id.frame_content);
        if (introFragment == null) introFragment = SettingsFragment.newInstance();
        getSupportFragmentManager().beginTransaction().replace(R.id.frame_content, introFragment).commit();
    }

    @Override
    public ManualInjector<Fragment> supportFragmentInjector() {
        return componentSource;
    }
}

File: app/src/main/java/eu/darken/bluemusic/settings/ui/SettingsActivity.kt
package eu.darken.bluemusic.settings.ui

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import eu.darken.bluemusic.R
import eu.darken.bluemusic.settings.ui.general.SettingsFragment
import eu.darken.bluemusic.util.ActivityUtil
import eu.darken.mvpbakery.base.MVPBakery.Companion.builder
import eu.darken.mvpbakery.base.ViewModelRetainer
import eu.darken.mvpbakery.injection.ComponentSource
import eu.darken.mvpbakery.injection.InjectedPresenter
import eu.darken.mvpbakery.injection.ManualInjector
import eu.darken.mvpbakery.injection.PresenterInjectionCallback
import eu.darken.mvpbakery.injection.fragment.HasManualFragmentInjector
import javax.inject.Inject

class SettingsActivity : AppCompatActivity(), SettingsActivityPresenter.View, HasManualFragmentInjector {
    @Inject lateinit var componentSource: ComponentSource<Fragment>
    override fun onCreate(savedInstanceState: Bundle?) {
        setTheme(R.style.BaseAppTheme)
        super.onCreate(savedInstanceState)
        builder<SettingsActivityPresenter.View, SettingsActivityPresenter>()
                .presenterFactory(InjectedPresenter(this))
                .presenterRetainer(ViewModelRetainer(this))
                .addPresenterCallback(PresenterInjectionCallback(this))
                .attach(this)
        setContentView(R.layout.activity_layout_settings)
    }

    override fun showSettings() {
        var introFragment = supportFragmentManager.findFragmentById(R.id.frame_content)
        if (introFragment == null) introFragment = SettingsFragment.newInstance()
        supportFragmentManager.beginTransaction().replace(R.id.frame_content, introFragment!!).commitAllowingStateLoss()
    }

    override fun supportFragmentInjector(): ManualInjector<Fragment> = componentSource

    override fun startActivityForResult(intent: Intent, requestCode: Int, options: Bundle?) {
        ActivityUtil.tryStartActivity(this) { super@SettingsActivity.startActivityForResult(intent, requestCode, options) }
    }
}